### PR TITLE
Add nullability to ref types; removal of all cbs now removes dict entry

### PR DESCRIPTION
**Pull request by coalstevens**

### CODE REVIEW

**Comment by justinawrey** [comment]
Thanks! I will get to this a bit later

File: Runtime/Reactive.cs [hunk 1]
```diff
@@ -5,18 +5,19 @@
 namespace ReactiveUnity
 {
     [Serializable]
+    #nullable enable
```
Comment (justinawrey) [inline Runtime/Reactive.cs:8]: Can we put the `#nullable enable` above the namespace declaration?

File: Runtime/Reactive.cs [hunk 2]
```diff
@@ -65,26 +73,33 @@ private void Set(T to)
...
+            if(_predicates[predicate].Count == 1) 
+            {
+                _predicates.Remove(predicate);
+                return;
+            }
+            
             _predicates[predicate].Remove(cb);
         }
     }
+    #nullable disable
```
Comment (justinawrey) [inline Runtime/Reactive.cs:104]: Same here, move the `#nullable disable` to under the last closing bracket

File: Runtime/Reactive.cs [hunk 3]
```diff
@@ -28,33 +29,40 @@ public T Value
...
+            
+            if (prevValue != null) {
+                if (prevValue.Equals(_val))
+                {
+                    return;
+                }
+            }
 
-            if (prevValue.Equals(_val))
+            if (prevValue == null && _val == null)
```
Comment (justinawrey) [inline Runtime/Reactive.cs:57]: This doesn't work if you want to set the value of the reactive variable back to `null` (which I presume is a valid use-case).

I think all we need to do is change the `prevValue.Equals(_val)` to `Equals(prevValue, _val)` (see [Object.Equals](https://learn.microsoft.com/en-us/dotnet/api/system.object.equals?view=net-8.0#system-object-equals(system-object-system-object)); its basically just a version of an equality check that supports `null`)

**Review by justinawrey — CHANGES_REQUESTED** [review]

File: Runtime/Reactive.cs [hunk 4]
```diff
@@ -65,26 +73,33 @@ private void Set(T to)
...
                 return;
             }
-
+            
+            if(_predicates[predicate].Count == 1) 
+            {
+                _predicates.Remove(predicate);
+                return;
+            }
+
```
[the remark this answers is not in the document]
Comment (coalstevens) [reply Runtime/Reactive.cs:100]: Nice I agree that's cleaner

**Pull request merged**